beacon-chain/core/helpers/validators.go

```diff
@@ -271,13 +296,18 @@ func BeaconProposerIndex(ctx context.Context, state state.ReadOnlyBeaconState) (
 			return 0, err
 		}
 		if r != nil && !bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
-			proposerIndices, ok := proposerIndicesCache.ProposerIndices(wantedEpoch+1, bytesutil.ToBytes32(r))
-			if ok {
-				return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
+			pid, err := cachedProposerIndexAtSlot(slot, [32]byte(r))
+			if err == nil {
+				return pid, nil
 			}
-			if err := UpdateProposerIndicesInCache(ctx, state, time.CurrentEpoch(state)); err != nil {
+			if err := UpdateProposerIndicesInCache(ctx, state, e); err != nil {
 				return 0, errors.Wrap(err, "could not update committee cache")
 			}
+			// This should not error except at genesis
```

Comment:
This will not error in genesis either because we skip the original `if` for genesis epoch